### PR TITLE
center signup form on home page

### PR DESCRIPTION
**Pull request by kne42**
([relevant issue](https://airtable.com/tblNxANme1LbVQqq1/viwISga6FERUatih1/recj5Qm8KyA5hCXvv?blocks=hide))

Screen Shot 2021-06-25 at 1 58 25 PM

### CODE REVIEW

**Review by codemonkey800 — APPROVED** [review]
LGTM 🥔

File: client/src/components/SignupForm/SignupForm.tsx [hunk 1]
```diff
@@ -53,7 +55,11 @@ export function SignupForm({ onSubmit }: Props) {
   };
 
   return (
-    <ColumnLayout className="bg-napari-light p-6 screen-495:p-12">
+    <ColumnLayout
+      className="bg-napari-light p-6 screen-495:p-12"
+      // Use 3-column layout instead of 4-column layout.
+      classes={isHome ? { fourColumn: 'screen-1150:grid-cols-napari-3' } : {}}
```
Comment (codemonkey800) [inline client/src/components/SignupForm/SignupForm.tsx:61]: unrelated to the PR, but how do you feel about the ColumnLayout API? i feel iffy about it, but I haven't thought about cleaner looking override it 😆

Comment (kne42) [reply client/src/components/SignupForm/SignupForm.tsx:61]: it's a little bit confusing having both `className` and `classes` as options. at minimum I would rename `classes` to `overrides` or something like that

I also think the component may be too generic with this and would probably be fine with a `variant` version instead

File: client/src/components/SignupForm/SignupForm.tsx [hunk 2]
```diff
@@ -53,7 +55,11 @@ export function SignupForm({ onSubmit }: Props) {
...
+    >
```
Comment (kne42) [inline client/src/components/SignupForm/SignupForm.tsx:62]: in terms of the actual API though I think the following appeals to me more:
```suggestion
<ColumnLayout className={clsx('bg-napari-light p-6 screen-495:p12', isHome && '~screen-1150:grid-cols-napari-4 screen-1150:grid-cols-napari-3')} />
```

basically `~` before a class name signifies that it should be removed

File: client/src/components/SignupForm/SignupForm.tsx (hunk 1 above)
Comment (justinelarsen) [reply client/src/components/SignupForm/SignupForm.tsx:61]: I agree with @kne42 - having both with close names is confusing.

**Review by justinelarsen — APPROVED** [review]
Code looks good, but difficult to test without a UI preview link

File: client/src/components/SignupForm/SignupForm.tsx (hunk 1 above)
Comment (codemonkey800) [reply client/src/components/SignupForm/SignupForm.tsx:61]: It's meant to be consistent with how Material UI handles overrides: https://material-ui.com/customization/components/

We can change it for this component, but the `classes` and `className` API isn't going away completely (at least for Material UI components)

**Comment by kne42** [comment]
@justinelarsen you can now test it out at this url: https://kira-sandbox-frontend.dev.imaging.cziscience.com

File: client/src/components/SignupForm/SignupForm.tsx (hunk 2 above)
Comment (codemonkey800) [reply client/src/components/SignupForm/SignupForm.tsx:62]: I can see this working, although if we're going to use a special operators, we might as well use the important operator `!`:

```tsx
<ColumnLayout className={clsx(
  'bg-napari-light p-6 screen-495:p12',
  isHome && 'screen-1150:!grid-cols-napari-3'
)} />
```

But using `!important` is generally not recommended. Either way, this is good feedback and is something we can possibly look into during the quality sprint.

**Comment by justinelarsen** [comment]
Thanks @kne42 ! LGTM!

**Review by liaprins-czi — APPROVED** [review]
Looks great!

**Pull request merged**